drivers/volume/csi/csi.go

```diff
@@ -0,0 +1,1214 @@
+	VolumeSnapshotContents map[string]*kSnapshotv1beta1.VolumeSnapshotContent `json:"volumeSnapshotContents"`
+	VolumeSnapshotClasses  map[string]*kSnapshotv1beta1.VolumeSnapshotClass   `json:"volumeSnapshotClasses"`
+}
+
+//  GetVolumeSnapshotContent retrieves a backed up volume snapshot
+func (cbo *csiBackupObject) GetVolumeSnapshot(snapshotID string) (*kSnapshotv1beta1.VolumeSnapshot, error) {
+	vs, ok := cbo.VolumeSnapshots[snapshotID]
+	if !ok {
+		return nil, fmt.Errorf("failed to retrieve volume snapshot for snapshotID %s", snapshotID)
+	}
+	return vs, nil
+}
+
+// GetVolumeSnapshotContent retrieves a backed up volume snapshot content
+func (cbo *csiBackupObject) GetVolumeSnapshotContent(snapshotID string) (*kSnapshotv1beta1.VolumeSnapshotContent, error) {
+	vsc, ok := cbo.VolumeSnapshotContents[snapshotID]
+	if !ok {
+		return nil, fmt.Errorf("failed to retrieve volume snapshot content for snapshotID %s", snapshotID)
+	}
+	return vsc, nil
+}
+
+// GetVolumeSnapshotClass retrieves a backed up volume snapshot class
+func (cbo *csiBackupObject) GetVolumeSnapshotClass(snapshotID string) (*kSnapshotv1beta1.VolumeSnapshotClass, error) {
+	vs, ok := cbo.VolumeSnapshots[snapshotID]
+	if !ok {
+		return nil, fmt.Errorf("failed to retrieve volume snapshot for snapshotID %s", snapshotID)
+	}
+
+	if vs.Spec.VolumeSnapshotClassName == nil {
+		return nil, fmt.Errorf("failed to retrieve volume snapshot class for snapshot %s. Volume snapshot class is undefined", snapshotID)
+	}
+	vsClassName := *vs.Spec.VolumeSnapshotClassName
+
+	vsClass, ok := cbo.VolumeSnapshotClasses[vsClassName]
+	if !ok {
+		return nil, fmt.Errorf("failed to retrieve volume snapshot class for snapshotID %s", snapshotID)
+	}
+
+	return vsClass, nil
+}
+
+type csi struct {
+	snapshotClient *kSnapshotClient.Clientset
+
+	storkvolume.ClusterPairNotSupported
+	storkvolume.MigrationNotSupported
+	storkvolume.GroupSnapshotNotSupported
+	storkvolume.ClusterDomainsNotSupported
+	storkvolume.CloneNotSupported
+	storkvolume.SnapshotRestoreNotSupported
+}
+
+func (c *csi) Init(_ interface{}) error {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	cs, err := kSnapshotClient.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	c.snapshotClient = cs
+
+	return nil
+}
+
+func (c *csi) String() string {
+	return storkCSIDriverName
+}
+
+func (c *csi) Stop() error {
+	return nil
+}
+
+func (c *csi) OwnsPVC(pvc *v1.PersistentVolumeClaim) bool {
+	// Try to get info from the PV since storage class could be deleted
+	pv, err := core.Instance().GetPersistentVolume(pvc.Spec.VolumeName)
+	if err != nil {
+		log.PVCLog(pvc).Warnf("error getting pv %v for pvc %v: %v", pvc.Spec.VolumeName, pvc.Name, err)
+		return false
+	}
+	return c.OwnsPV(pv)
+}
+
+func (c *csi) HasNativeVolumeDriverSupport(pv *v1.PersistentVolume) bool {
+	return pv.Spec.CSI.Driver == snapv1.PortworxCsiProvisionerName ||
```

Comment:
Instead of having a hardcoded list here, it would be best for each driver to register their csi driver name to the volume driver. Then this can just make a call to get all the supported drivers and look at the list.

Can be done in a separate PR